src/plugins/curator/screens/teaser/Form.jsx

```diff
@@ -50,14 +50,14 @@ const getFieldsComponent = (type) => {
   }
 };
 
-const Form = ({ node, tab, type, isEditMode }) => {
+const Form = ({ form, node, tab, type, isEditMode }) => {
   const [TeaserFields, setTeaserFields] = useState(null);
   const streamId = StreamId.fromString(`${node.schema().getCurie().getMessage()}.history:${node.get('_id')}`);
```

Comment:
I don't know what you mean by creating it's own stream. Example?

Reply:
@rsumilang right now all these Form components are making their own streamIds: https://github.com/triniti/cms-js/blob/5bcd11804a7f9561b5a05946505beae199dfc7b2/src/plugins/iam/screens/user/Form.jsx#L16

but if we are going to pass the node into the History component then the History component can create the streamId inside itself like so

```
const streamId = StreamId.fromString(`${node.schema().getCurie().getMessage()}.history:${node.get('_id')}`);
```

and then the Forms dont have to worry about it, only the History component does it

Reply:
like it could happen inside the delegate because that delegate will have access to the node via ownProps https://github.com/triniti/cms-js/blob/master/src/plugins/pbjx/components/history/delegate.js#L14